TFUtil.py

```diff
@@ -1028,10 +1028,11 @@ def get_placeholder_flattened(self, keep_dims=False):
       ndim -= 1
     if len(dyn_axes) > 1:
       shape = tf.shape(x)
+      shape_as_list = x._shape_as_list()
       x = tf.reshape(
         x,
         [tf.reduce_prod([shape[i] for i in dyn_axes])] +
-        [shape[i] for i in range(ndim) if i not in dyn_axes])
+        [shape_as_list[i] for i in range(ndim) if i not in dyn_axes])
```

Comment:
Why? The old could was just the same.

Reply:
Technically it was the same, but `[shape[i] for i in range(ndim) if i not in dyn_axes]` consists of tensors and returns None shape for static dims. However it's possible to specify the static dims by gettign the values from `x.get_shape().as_list()`.

Reply:
No, `shape[i]` is never `None`. But `shape_as_list[i]` could be `None`, so this would be wrong.

Reply:
`shape[i]` is not `None` but it's a tensor. After `tf.reshape` with a tensor, the resulting dim becomes None even it was defined before.

Reply:
Yes, why does that matter? This is the argument for `tf.reshape`, and it must not be `None`.

Reply:
The result of `tf.reshape` with Tensor as an argument gives `None`. Even for `dim`.

Reply:
This is not the result of `tf.reshape`, this is the argument to `tf.reshape`. For that argument, you must not give `None`.

Reply:
Maybe you misunderstand me. To make it clear:
Sth like this is allowed:

    tf.reshape(x, (1, 2, 3, some_tensor, 4, 5))

Sth like this is **not** allowed (i.e. `None` is not allowed):

    tf.reshape(x, (1, 2, None, some_tensor, 4, 5))

Reply:
I know, that this is not a result of `tf.reshape`. My point is: why should we pass a tensor as an argument (smth like `tf.reshape(x, (1, 2, 3, some_tensor, 4, 5))`), when we could explicitly pass a number there (`tf.reshape(x, (1, 2, 3, 34, 4, 5))`))?

Reply:
`shape_as_list[i]` can be `None`. You must not pass `None`.

Reply:
Yes, it can be. But `[shape_as_list[i] for i in range(ndim) if i not in dyn_axes]` doesn't have `None`. because these are such elements, which are `not in dyn_axes`.

Reply:
Do not rely on that. That does not always have to be the case. `shape_as_list[i]` could be `None`. E.g. the feature dim can be None, and is not part of `dyn_axes` here. But there could be other cases. Even if this works now, for some of the cases, this might break in the future. Don't write code like that.